plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -1026,6 +1026,51 @@ fu_genesys_tsdigit_value(gchar c)
 	return g_ascii_digit_value(c);
 }
 
+static gboolean
+fu_genesys_usbhub_device_validate_token(FuGenesysUsbhubDevice *self, GError **error)
+{
+	const gchar *valid_tokens[] = {"GL 3.0 Hub", "GL 3.1 Hub", NULL};
+	g_autofree gchar *token = NULL;
+	g_autoptr(GBytes) token_blob = NULL;
+	g_autoptr(GBytes) token_blob_trunc = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* get 0x80 string descriptor */
+	token_blob = g_usb_device_get_string_descriptor_bytes_full(
+	    fu_usb_device_get_dev(FU_USB_DEVICE(self)),
+	    0x80,
+	    G_USB_DEVICE_LANGID_ENGLISH_UNITED_STATES,
+	    64,
+	    &error_local);
+	if (token_blob == NULL) {
+		if (g_error_matches(error_local, G_USB_DEVICE_ERROR, G_USB_DEVICE_ERROR_IO)) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "cannot get 0x80 string descriptor: %s",
+				    error_local->message);
```

Comment:
Why create a new error?  Just steal the pointer.

Reply:
It's the wrong GQuark error domain -- it needs to be either FWUPD_ERROR_NOT_SUPPORTED or G_IO_ERROR_NOT_SUPPORTED to avoid the error.